qfieldsync/core/cloud_api.py

```diff
@@ -610,6 +610,20 @@ def projects(self) -> Optional[List[CloudProject]]:
     def error_reason(self) -> str:
         return self._error_reason
 
+    @property
+    def is_currently_open_project_cloud_local(self) -> bool:
+        project_dir = QgsProject.instance().homePath()
+
+        found = False
+        for project_id, local_dir in self.preferences.value(
+            "qfieldCloudProjectLocalDirs"
+        ).items():
+            if local_dir == project_dir:
+                found = True
+                break
```

Comment:
```suggestion
        local_project_dirs = self.preferences.value("qfieldCloudProjectLocalDirs").items()
        found = any((project_dir == local_dir for _, local_dir in local_project_dirs))
```

Some python yoga?
Your call which version is more readable